q2_fmt/_peds.py

```diff
@@ -530,28 +532,79 @@ def peds_simulation(table: pd.DataFrame, metadata: qiime2.Metadata,
                              " and needs more than one recipient"
                              " to successfully shuffle.")
 
+    if not peds_rarefaction:
+        num_resamples = 1
+
     peds = sample_peds(
            table=table, metadata=metadata,
            time_column=time_column,
            reference_column=reference_column,
            subject_column=subject_column,
-           filter_missing_references=filter_missing_references
-           ).set_index("id")
-    actual_peds = peds["measure"]
+           filter_missing_references=filter_missing_references,
+           num_resamples=num_resamples,
+           sampling_depth=sampling_depth
+           ).set_index('id')
+
+    actual_peds = peds['measure']
+    # resetting attrs because they get wipe with set_index
+    peds['measure'].attrs.update({
+        'title': "Sample PEDS",
+        'description': 'Proportional Engraftment of Donor Strains'
+    })
+    peds['group'].attrs.update({
+        'title': time_column,
+        'description': 'Time'
+    })
+    peds['subject'].attrs.update({
+        'title': subject_column,
+        'description': 'Subject IDs linking samples across time'
+    })
+    peds['transfered_donor_features'].attrs.update({
+        'title': "Transfered Reference Features",
+        'description': '...'
+    })
+    peds['total_donor_features'].attrs.update({
+        'title': "Total Reference Features",
+        'description': '...'
+    })
+    peds['donor'].attrs.update({
+        'title': reference_column,
+        'description': 'Donor'
+    })
 
     # Mismatch simulation:
-    table = table > 0
     recip_df = _create_recipient_table(used_references, metadata_df, table)
     donor_df = table[table.index.isin(used_references)]
     mismatched_df = \
         _create_mismatched_pairs(recip_df,
                                  metadata_df,
                                  used_references,
                                  reference_column)
-    duplicated_recip_table = _create_duplicated_recip_table(mismatched_df,
-                                                            recip_df)
-    donor_mask = _create_sim_masking(mismatched_df, donor_df, reference_column)
-    recip_mask = _mask_recipient(donor_mask, duplicated_recip_table)
+    simulated_mismatched_df = _simulate_mismatched_pairs(mismatched_df,
+                                                         num_resamples)
+
+    simulated_recip_table, simulated_donor_table =\
+        _create_duplicated_tables(simulated_mismatched_df, recip_df,
+                                  donor_df)
+    # concating or recip and donor tables so column number stays the same after
+    # subsampling
+    simulated_table = pd.concat([simulated_recip_table,
+                                 simulated_donor_table])
+    rarefied_simulated_table = _subsample(table=simulated_table,
```

Comment:
this will need most of what _subsample does (note for when it is removed)

Reply:
We have decided that this isn't going to work